lightning_uq_box/models/masked_ensemble/utils.py

```diff
@@ -23,44 +23,41 @@ def convert_deterministic_to_masked_ensemble(
             the interval [1, 6].
     """
     for name, value in list(deterministic_model._modules.items()):
-        if deterministic_model._modules[name]._modules:
+        assert value
+        if value._modules:
             convert_deterministic_to_masked_ensemble(
-                deterministic_model._modules[name],
-                num_estimators=num_estimators,
-                scale=scale,
+                value, num_estimators=num_estimators, scale=scale
             )
-        elif "Conv2d" in deterministic_model._modules[name].__class__.__name__:
-            curr_layer = deterministic_model._modules[name]
-            if curr_layer.in_channels >= 10:
+        elif isinstance(value, nn.Conv2d):
+            if value.in_channels >= 10:
```

Comment:
Also I cannot remember anymore why the docstring says 10 input channels. but that is wrong, since the method works also for inputs with less than 10 channels

Reply:
oh wait, maybe it's also correct, because the changes should only be applied to deeper layers. Will check the original repo again...